### ISSUE TITLE
vs2017 building issue

### ISSUE THREAD

**Issue by Manimozhis95** [zeromq/czmq#1780]
Any one build successfully CZMQ using vs2017

**Comment by bluca** [comment]
@evoskuil @somdoron @sigiesec does any of you guys know about VS and CZMQ? There's been questions on the mailing list as well

**Comment by sigiesec** [comment]
No, sorry, I have never used czmq.

**Comment by Formulator** [comment]
Sorry I haven't VS2017 installed, but I’ve been trying to build CZMQ on Windows 10 using VS2015, the build fails after a few seconds and I get errors; https://pastebin.com/a2r2KJi0 

This is what I’ve done so far;

git clone --depth 1 -b stable https://github.com/jedisct1/libsodium.git
cd libsodium\builds\msvc\build
buildall.bat
cd ..\..\..\..

:: if libsodium is on disk, the Windows build of libzmq will automatically use it
git clone git://github.com/zeromq/libzmq.git
cd libzmq\builds\msvc
configure.bat
cd build
buildall.bat
cd ..\..\..\..

git clone git://github.com/zeromq/czmq.git
cd czmq\builds\msvc
configure.bat
cd vs2015
build.bat

Would love to get CZMQ building smoothly on Windows!

**Comment by myd7349** [comment]
I tried to build czmq using VS2017 on Win10 about one and a half months ago. The auxiliary tools I used is [zproject](https://github.com/zeromq/zproject) and [gsl](https://github.com/imatix/gsl). Of cause, I installed them on Win10 via `WSL(Windows Subsystem for Linux)`.

**Comment by somdoron** [comment]
Not really, never built it on windows.
I can give it a try next week.

**Comment by sgluss** [comment]
@myd7349 did you try the step specified by jacktrades52 in this issue: https://github.com/zeromq/czmq/issues/1617

This worked for me, got CZMQ built on Win10/VS2017

I also had to:
	-switch all the solutions to x64 config for libzmq/czmq
	-make sure build targets match for libzmq/czmq

**Comment by myd7349** [comment]
@sgluss No, I didn't try it yet. I will give it a try later.

**Comment by Formulator** [comment]
Nice work @myd7349 , vcpkg sounds like an awesome time-saver!  When you get czmq ported I'd like to give it a try building my Windows / Linux app which uses boost, czmq & zeromq.  One question I have is:  Typically I normally build zeromq using libsodium as the crypto library because I discovered throughput performance issues with the default tweetnacl.  How would I install zeromq within vcpkg making sure that libsodium was used for crypto?  Thanks for any clarification.

**Comment by myd7349** [comment]
@Formulator Thanks for your kindly reply. As far as I know, `zeromq` port in `vcpkg` doesn't link to `libsodium`. I also created a PR for using `zeromq` with `libsodium`: https://github.com/Microsoft/vcpkg/pull/4214 .

Once merged, we can reinstall `zeromq` like this:
```bash
vcpkg install zeromq[libsodium]
```
Even if this PR was not accepted, we may modify local zeromq portfile as what I did in the above PR.

**Comment by myd7349** [comment]
Hi! @Formulator Good news! I finally finish this work. Now, we can install `czmq + zeromq + libsodium` in `vcpkg` like this:
> vcpkg install zeromq[sodium]
vcpkg install czmq

**Comment by Formulator** [comment]
Wow! @myd7349 thanks, I compiled & ran my app on Windows 10 using Visual Studio 2019 with czmq via vcpkg.  So easy, fast and simple..... Except for one issue:  

`Error CMake Error at vcpkg/installed/x86-windows/share/czmq/czmqConfig.cmake:31 (if):
  if given arguments:
    "AND" "FALSE"
  Unknown arguments specified vcpkg/installed/x86-windows/share/czmq/czmqConfig.cmake	31
`

Not sure what the logic at `czmqConfig.cmake` line 31 intends:

`find_dependency(ZeroMQ)`
`if ( AND FALSE)`
`    find_dependency(OpenSSL)`
`    find_dependency(ZLIB)`
`endif ()`


But by commenting out the `if` & `endif` lines I bypassed the error and got it to function.
